tests/test_utils.py

```diff
@@ -136,15 +136,44 @@ async def mutate_registrations():
     assert cb4.exec_count == 1  # Registered during previous invoke(), was called
 
 
+# Timeout within 2 seconds (not)
+@pytest.mark.timeout(2)
 def test_random_port():
     assert random_port(9000, 9000) == 9000
 
+    # Starting port
+    port = 9001
+    # Test `k` continguous ports
+    k = 10
+    # Number of times to try to find a port range
+    n = 100
+
+    # Find a range of `k` ports that are all available
+    for i in range(n):
+        assert i < n - 1, f"Could not find {k} continguous ports to use for testing"
```

Comment:
Would be a bit clearer to have this test after the `for` loop (but with `n` instead of `n-1`, I think)